src/hotel_reservations/data_processor.py

```
@@ -0,0 +1,118 @@
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import current_timestamp, to_utc_timestamp
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.model_selection import train_test_split
+
+from hotel_reservations.config import ProjectConfig
+
+
+class DataProcessor(BaseEstimator, TransformerMixin):
+    """Class to preprocess and split data for the Hotel Reservations project."""
+
+    def __init__(self, config: ProjectConfig, fe_features: list = []):
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)